cmd/main.go

```diff
@@ -94,6 +97,23 @@ func main() {
 		TLSOpts: tlsOpts,
 	})
 
+	defaultNamespaces := map[string]cache.Config{}
+	operatorNamespace, err := getOperatorNamespace()
+	if err != nil {
+		setupLog.Error(err, "manager requires namespace to be registered for controllers to reconcile")
+		os.Exit(1)
+	}
+	// ensure we always cache items from operator namespace
+	defaultNamespaces[operatorNamespace] = cache.Config{}
```

Comment:
how will this change work where the operator should be able to manage the csi drivers in all the namespaces?  https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md#goals-and-guidelines

Reply:
in my tests probably csi driver was already deployed, do you know how rook does it?

Reply:
Rook isn't doing anything explicitly for cluster scoped resources and based on my testing in https://github.com/red-hat-storage/ocs-operator/pull/2911 we do get events for them, ocs-op only watches 2 namespaces.

Reply:
retested, it's working w/o any changes
```
> date; k get csidriver  -w
Fri Jan  3 06:26:07 UTC 2025
NAME               ATTACHREQUIRED   PODINFOONMOUNT   STORAGECAPACITY   TOKENREQUESTS   REQUIRESREPUBLISH   MODES        AGE
rbd.csi.ceph.com   true             false            false             <unset>         false               Persistent   64s
rbd.csi.ceph.com   true             false            false             <unset>         false               Persistent   87s
rbd.csi.ceph.com   true             false            false             <unset>         false               Persistent   0s

> date; k delete csidriver rbd.csi.ceph.com; date
Fri Jan  3 06:26:31 UTC 2025
csidriver.storage.k8s.io "rbd.csi.ceph.com" deleted
Fri Jan  3 06:26:31 UTC 2025

Reply:
its not for Rook, its for standalone deployment of csi driver

Reply:
No, I meant rook also deploys csidriver even if it is watching single namespace.